### PR TITLE
feat(lib: menubar): add paginated scrolling

### PR DESCRIPTION
**Pull request by actionless**

### CODE REVIEW

File: lib/menubar/init.lua.in [hunk 1]
```diff
@@ -280,9 +321,6 @@ end
 --- Get a menubar wibox.
 -- @return menubar wibox.
 function menubar.get()
-    if app_folders then
```
Comment (actionless) [inline lib/menubar/init.lua.in]: it's unset, mb it should be `if menubar.app_folders then` or smth like that

File: lib/menubar/init.lua.in [hunk 2]
```diff
@@ -55,6 +56,15 @@ menubar.geometry = { width = nil,
                      x = nil,
                      y = nil }
 
+--- Width of blank space left in the right side.
+menubar.right_margin = 50
+
+--- Label used for "Next page", default ">>".
+menubar.right_label = "&gt;&gt;"
```
Comment (blueyed) [inline lib/menubar/init.lua.in]: Can we use some fancy Unicode symbol here instead?

File: lib/menubar/init.lua.in [hunk 3]
```diff
@@ -55,6 +56,15 @@ menubar.geometry = { width = nil,
...
+
+--- Label used for "Previous page", default "<<".
+menubar.left_label = "&lt;&lt;"
```
Comment (blueyed) [inline lib/menubar/init.lua.in]: Same as above:

> Can we use some fancy Unicode symbol here instead?

**Comment by actionless** [comment]
which will work better? `⏩, ⏵, ▶`

**Comment by blueyed** [comment]
I like ⏩..

**Comment by actionless** [comment]
@blueyed the one you proposed was rendering strange on my other machine (probably too small default fontsize?) so tried "▶▶", it looks almost the same but more crispy

File: lib/menubar/init.lua.in [hunk 4]
```diff
@@ -16,9 +16,10 @@ local theme = require("beautiful")
 local wibox = require("wibox")
 
 -- menubar
-local menubar = { mt = {} }
+local menubar = { mt = {}, menu_entries = {} }
```
Comment (actionless) [inline lib/menubar/init.lua.in]: it was global

or better to make it just local instead?

**Comment by actionless** [comment]
number of commits started to grow so i can squash it after review if needed

**Comment by blueyed** [comment]
I am getting this error when invoking the menubar:

    W: awesome: luaA_dofunction:77: error while running function
    stack traceback:
            /usr/local/share/awesome/lib/menubar/init.lua:137: in function 'get_current_page'
            /usr/local/share/awesome/lib/menubar/init.lua:225: in function 'menulist_update'
            /usr/local/share/awesome/lib/menubar/init.lua:311: in function 'show'
            /home/user/.config/awesome/rc.lua:1142: in function 'press'
            /usr/local/share/awesome/lib/awful/key.lua:42: in function </usr/local/share/awesome/lib/awful/key.lua:42>
    error: /usr/local/share/awesome/lib/menubar/init.lua:137: attempt to perform arithmetic on field 'width' (a nil value)

File: lib/menubar/init.lua.in (hunk 4 above)
Comment (blueyed) [inline lib/menubar/init.lua.in]: Seems to be OK like this.  The entries belong to the menubar (instance) AFAICS.

File: lib/menubar/init.lua.in (hunk 1 above)
Comment (blueyed) [inline lib/menubar/init.lua.in]: It was never used in awesome itself, and was probably used to override / provide it from outside.

Given this, it should probably be used via `menubar.app_folders`, but maybe only when it's being requested again..

**Comment by blueyed** [comment]
:+1: 
Please squash the commits.

**Comment by actionless** [comment]
done

File: lib/menubar/init.lua.in (hunk 1 above)
Comment (actionless) [inline lib/menubar/init.lua.in]: anyway it still can be used as `menubar.menu_gen.all_menu_dirs` from outside

**Comment by blueyed** [comment]
Are you sure about the commit message, or do you want to adjust it?
I could do so also when merging it, but you probably do not want to use all the messages from the squashed commits as-is..?!

**Comment by actionless** [comment]
updated commit message

**Comment by blueyed** [comment]
Merged.  Thanks!

**Pull request merged**